packages/table/src/common/utils.ts

```diff
@@ -200,15 +200,25 @@ export const Utils = {
     },
 
     /**
-     * Partial shallow comparison between objects using the given list of keys.
+     * Shallow comparison between objects. If `keys` is provided, just that subset of keys will be
+     * compared; otherwise, all keys will be compared.
      */
-    shallowCompareKeys(objA: any, objB: any, keys: string[]) {
-        for (const key of keys) {
-            if (objA[key] !== objB[key]) {
+    shallowCompareKeys(objA: any, objB: any, keys?: string[]) {
+        if (keys != null) {
+            return shallowCompareKeys(objA, objB, keys);
```

Comment:
how is this not an immediate infinite loop every time?

can we please rename the internal function? might be legit to use `_` prefix here.